### ISSUE TITLE
[BottomNavigation] How can I use with Svelte?

### ISSUE THREAD

**Issue by ohmaha** [nativescript-community/ui-material-components#427]
I am very new and I am interested to create a mobile application by Svelte NativeScript.
But I don't know how can I use the BottomNavigation.
Is this library support Svelte?

**Comment by dnicer3** [comment]
Unfortunately, I would like to have a solution for Svelte as well. As far as I know, since Nativescript 8.0 was released, components BottomNavigation and Tabs are not supported by officials anymore, only by the community. I have already tried to bypass this by using a regular Nativescript and registerNativeViewElement function, however unsuccessfully.

**Comment by farfromrefug** [comment]
@ohmaha @dnicer3 yes svelte needs a special component for this.
@halfnelson started working on this https://github.com/halfnelson/ui-material-components/commit/904c71ec5cc43f74dd77df99e7ec6f09357c4fdf
@halfnelson what s the status on this?

**Comment by NoelOConnell** [comment]
I got this working on Stackblitz Preview today with help on the NS Discord.
Here's how I set it up.

Install the plugin in package.json
`"@nativescript-community/ui-material-bottom-navigation": "^7.0.25"`
I'm using that version since a message on Discord said that was the version supported by ns preview

Register the plugin in  `app.ts`
<details><summary>app.ts</summary>

```javascript
import {
  BottomNavigation,
  TabContentItem,
  TabStrip,
  TabStripItem,
} from '@nativescript-community/ui-material-bottom-navigation';

.....

registerNativeViewElement('bottomNavigation', () => BottomNavigation);
registerNativeViewElement('tabStrip', () => TabStrip);
registerNativeViewElement('tabStripItem', () => TabStripItem);
registerNativeViewElement('tabContentItem', () => TabContentItem);

svelteNativeNoFrame(App, {});
```

</details>

Add the component to your `Page.svelte`
<details><summary>Page.svelte</summary>

```javascript
<script lang="ts">
let selectedTab = 1

$:console.log({selectedTab})
</script>

<page xmlns:mdt="@nativescript-community/ui-material-bottom-navigation">
<bottomNavigation bind:selectedIndex={selectedTab}>

	<!-- The bottom tab UI is created via TabStrip (the containier) and TabStripItem (for each tab)-->
	<tabStrip>
		<tabStripItem>
			<label text="Home" />			
		</tabStripItem>
		<tabStripItem class="special">
			<label text="Account" />			
		</tabStripItem>
		<tabStripItem class="special">
			<label text="Search" />			
		</tabStripItem>
	</tabStrip>

	<!-- The number of TabContentItem components should corespond to the number of TabStripItem components -->
	<tabContentItem>
		<gridLayout>
			<label text="Home Page" class="h2 text-center" />
		</gridLayout>
	</tabContentItem>
	<tabContentItem>
		<gridLayout>
			<label text="Account Page" class="h2 text-center" />
		</gridLayout>
	</tabContentItem>
	<tabContentItem>
		<gridLayout>
			<label text="Search Page" class="h2 text-center" />
		</gridLayout>
	</tabContentItem>

</bottomNavigation>
</page>

```
</details>

**Comment by dnicer3** [comment]
@NoelOConnell, as I said, I have already tried same and anyway I can not make it working... See the code and package below.
![изображение](https://user-images.githubusercontent.com/62773632/211364900-c84e8da1-78a0-476b-87ed-30ef47b7dadc.png)
![изображение](https://user-images.githubusercontent.com/62773632/211365006-3f01e165-8d52-463b-9c1f-07d7c14d211b.png)
![изображение](https://user-images.githubusercontent.com/62773632/211365618-0162d2ac-15b7-4d1f-bffb-4c549a46233c.png)

**Comment by NoelOConnell** [comment]
@dnicer3 Can you try changing the order that the components are registered. I remember seeing some other issue about the order needing to be correct for tabs.

```javascript
registerNativeViewElement('bottomNavigation', () => BottomNavigation);
registerNativeViewElement('tabStrip', () => TabStrip);
registerNativeViewElement('tabStripItem', () => TabStripItem);
registerNativeViewElement('tabContentItem', () => TabContentItem);
```

**Comment by dnicer3** [comment]
@NoelOConnell Unfortunately, nothing is changing. (

**Comment by NoelOConnell** [comment]
@dnicer3 I have the plugin installed as a dev dependency. Not sure if that makes a difference.

![image](https://user-images.githubusercontent.com/966495/211369304-ddf26641-4593-4082-a70e-b0d11e7b92d7.png)

**Comment by vallemar** [comment]
@NoelOConnell  can you send your link of Stackblitz?

**Comment by NoelOConnell** [comment]
@vallemar here's the playground I'm experimenting with.

https://stackblitz.com/edit/nativescript-stackblitz-templates-gvy41h

**Comment by dnicer3** [comment]
@NoelOConnell I have even changed it to devDependencies, but still nothing...

**Comment by NoelOConnell** [comment]
@dnicer3 I've created a bare example that you can use to compare to your own and see if there's any differences.

https://stackblitz.com/edit/nativescript-stackblitz-templates-t1fr9o

**Comment by dnicer3** [comment]
@NoelOConnell thank you for assistance, however, mine code is 100% same but not working ))

**Comment by NoelOConnell** [comment]
@dnicer3 Yeah, I tried to run it locally with the emulator and got the same error as you. It works fine in Stackblitz so it's something to do with how the Nativescript Stackblitz has additional dependencies installed.

I looked at the package.json for `ui-material-bottom-navigation` and saw it has dependencies list.

I updated the `package.json` to include them and it worked in the emulator locally.
Can you try adding these dependencies.
```
"@nativescript-community/ui-material-bottom-navigation": "^7.0.34",    
"@nativescript-community/ui-material-core-tabs": "^7.0.34",
"@nativescript/core": "~8.4.0",
```

**Comment by dnicer3** [comment]
@NoelOConnell you are the God, thanks. Finally, it's working!
